src/client/splitor.cpp

```diff
@@ -469,5 +467,19 @@ RequestSourceInfo Splitor::CalcRequestSourceInfo(IOTracker* ioTracker,
     return {};
 }
 
+bool Splitor::NeedGetOrAllocateSegment(MetaCacheErrorType error, OpType opType,
+                                       const ChunkIDInfo& chunkInfo,
+                                       const MetaCache* metaCache) {
+    if (error == MetaCacheErrorType::CHUNKINFO_NOT_FOUND) {
+        return true;
+    } else if (error == MetaCacheErrorType::OK && !chunkInfo.chunkExist) {
+        return (opType == OpType::WRITE) ||
```

Comment:
If a read-only node becomes a read-write node, we need to restart nedb? Here write chunk not exist does not getorallocate segment.

Reply:
> If a read-only node becomes a read-write node, we need to restart nebd? 

reopen or re-map is needed, but this code is nothing about read-only

> Here write chunk not exist does not getorallocate segment.

nope, if it's a write operation this function returns true.